insonmnia/hardware/cpu/device.go

```diff
@@ -1,27 +1,28 @@
 package cpu
 
 import (
-	"github.com/cnf/structhash"
+	"errors"
+
 	"github.com/shirou/gopsutil/cpu"
+	"github.com/sonm-io/core/proto"
 )
 
-// Device describes a CPU device.
-type Device cpu.InfoStat
-
-func GetCPUDevices() ([]Device, error) {
+func GetCPUDevice() (*sonm.CPUDevice, error) {
 	info, err := cpu.Info()
 	if err != nil {
 		return nil, err
 	}
 
-	devices := []Device{}
-	for _, device := range info {
-		dev := Device(device)
-		devices = append(devices, dev)
+	if len(info) == 0 {
+		// o_O
+		return nil, errors.New("no CPU detected")
+	}
+
+	dev := &sonm.CPUDevice{ModelName: info[0].ModelName}
```

Comment:
And here the description is required, it's not clear why you take the first model name for the entire device set.